File: example/src/main/java/ru/virgil/example/user/UserDetailsController.java
package ru.virgil.example.user;

import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequiredArgsConstructor
@RequestMapping("/user_details")
public class UserDetailsController {

    private final UserDetailsService userDetailsService;
    private final UserDetailsMapper userDetailsMapper;

    @GetMapping
    public UserDetailsDto get() {
        UserDetails currentUser = userDetailsService.getCurrentUser();
        return userDetailsMapper.toDto(currentUser);
    }

}

File: example/src/main/java/ru/virgil/example/user/UserDetailsController.kt
package ru.virgil.example.user

import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/user_details")
class UserDetailsController(
    private val userDetailsService: UserDetailsService,
) : UserDetailsMapper {

    @GetMapping
    fun get(): UserDetailsDto {
        val currentUser = userDetailsService.getCurrentUser()
        return currentUser.toDto()
    }
}
